app/src/main/java/com/onboarding/parkingsystemjava/utils/ReservationVerifier.java

```diff
@@ -0,0 +1,58 @@
+package com.onboarding.parkingsystemjava.utils;
+
+import com.onboarding.parkingsystemjava.database.ParkingDatabase;
+import com.onboarding.parkingsystemjava.entity.Reservation;
+import java.util.ArrayList;
+import java.util.Calendar;
+
+public class ReservationVerifier {
+    private final ParkingDatabase database;
+
+    public ReservationVerifier(ParkingDatabase database) {
+        this.database = database;
+    }
+
+    public boolean isReservationOK(Reservation reserve) {
+        Integer lot = reserve.getParkingLot();
+        if (database.getReservations().containsKey(lot)) {
+            ArrayList<Reservation> lotReservations = database.getReservations().get(lot);
+            return checkArrayList(lotReservations, reserve);
+        } else {
+            return true;
+        }
+    }
+
+    private boolean isOverlapping(Reservation reservation, Reservation anotherReservation) {
+        Calendar reservationStartDate = reservation.getStartDate();
+        Calendar reservationEndDate = reservation.getEndDate();
+        Calendar anotherReservationStartDate = anotherReservation.getStartDate();
+        Calendar anotherReservationEndDate = anotherReservation.getEndDate();
+
+        //start or full overlap
+        if (reservationStartDate.before(anotherReservationStartDate) &&
+                reservationEndDate.after(anotherReservationStartDate)) {
+            return true;
+        } else {
+            //end overlap
+            if (reservationStartDate.before(anotherReservationEndDate) &&
+                    reservationEndDate.after(anotherReservationEndDate)) {
+                return true;
+            } else {
+                //stored reservation full overlaps new reservation
+                return reservationStartDate.after(anotherReservationStartDate) &&
+                        reservationEndDate.before(anotherReservationEndDate);
+            }
+        }
+    }
+
+    private boolean checkArrayList(ArrayList<Reservation> arrayList, Reservation reservation) {
+        Reservation storedReservation;
+        for (int i = ConstantUtils.INT_ZERO; i < arrayList.size(); i++) {
```

Comment:
can use .foreach

Reply:
requieres apimin 24, and the project has apimin 23